src/lib/structures/Logger.ts

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function */
+
+import EventEmitter from 'events';
+import JSLogger, { ILogger as JSILogger, ILoggerOpts, ILogHandler, ILogLevel } from 'js-logger';
+import { Util } from '../util/Util';
+import type { GlobalLogger } from 'js-logger';
+import type { AutocompleteContext } from './contexts/AutocompleteContext';
+import type { CommandContext } from './contexts/CommandContext';
+import type { ComponentContext } from './contexts/ComponentContext';
+import type { Command } from './Command';
+import type { Component } from './Component';
+import type { Listener } from './Listener';
+import type { Plugin } from './Plugin';
+
+export enum LoggerEvents {
+	'HANDLER_RUN' = 'handlerRun',
```

Comment:
Should this not be specific to the type of run? Command, component, etc

Reply:
You can use x.isCommand(), isButton(), etc

Reply:
I still think it's cleaner to split this up. Although that will work too